### ISSUE TITLE
MissingJavadocMethod should consider nested class

### ISSUE THREAD

**Issue by Zustin** [checkstyle/checkstyle#11568]
I have read check documentation: https://checkstyle.sourceforge.io/config_javadoc.html#MissingJavadocMethod
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="MissingJavadocMethod">
    <property name="allowedAnnotations" value="Override,Deprecated"/>
    </module>
  </module>
</module>
        
/var/tmp $ cat Test.java
public class Test {
  class subTest {
    public void test() {}  // should report a warning here
  }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java -jar checkstyle-10.1-all.jar -c ./cs.config Test.java
Starting audit...
Audit done.
```

---
**Describe what you expect in detail.**
---
Hi, I found a false negative about this rule. CheckStyle should have reported a warning in line 3 and this FN is caused by the inner class `subTest`. Thanks for your consideration.

**Comment by romani** [comment]
https://checkstyle.sourceforge.io/config_javadoc.html#MissingJavadocMethod_Properties

Default scope is public. Class is not public, so method is also not visible outside of outer class. So it will not be in html pages of jdk javadoc tool.

**Comment by Zustin** [comment]
Got it, thanks for your respones.
